### PR TITLE
Update target frameworks

### PR DESCRIPTION
**Pull request by NeilSchiefer**
## Description

Updated target frameworks to support only net6.0, netcoreapp3.1, and net48.

Added editorconfig and Directory.Build.props for consistency

## Type of change: 
This is a breaking change and version will be updated to 2.0.0.

---

<sup>_[Reviewer guidelines](../blob/main/CONTRIBUTING.md#reviewing-changes)_</sup>

### CODE REVIEW

**Review by eslutz — APPROVED** [review]
Looks good to me :thumbsup:

**Review by JasonBock — COMMENTED** [review]
I'm wondering if we create a GC "helper" class here in our test class, that may make it easier. Like a `GCNoRegion` class that implements IDisposable. On construction, we `TryStartNoGCRegion`, and if that's `false`, we throw an exception. In `Dispose()`, it calls `EndNoGCRegion()`. Then devs could simply do `using new GCNoRegion(...)` and that's it.

Just a thought

**Comment by NeilSchiefer** [comment]
> I'm wondering if we create a GC "helper" class here in our test class, that may make it easier. Like a `GCNoRegion` class that implements IDisposable. On construction, we `TryStartNoGCRegion`, and if that's `false`, we throw an exception. In `Dispose()`, it calls `EndNoGCRegion()`. Then devs could simply do `using new GCNoRegion(...)` and that's it.
> 
> Just a thought

Ah yeah that would clean up a lot of code.  Going to give it a shot.

**Comment by NeilSchiefer** [comment]
> > I'm wondering if we create a GC "helper" class here in our test class, that may make it easier. Like a `GCNoRegion` class that implements IDisposable. On construction, we `TryStartNoGCRegion`, and if that's `false`, we throw an exception. In `Dispose()`, it calls `EndNoGCRegion()`. Then devs could simply do `using new GCNoRegion(...)` and that's it.
> > Just a thought
> 
> Ah yeah that would clean up a lot of code. Going to give it a shot.

@JasonBock  Can you let me know if this makes sense to do it like this - https://github.com/RockLib/RockLib.Reflection.Optimized/pull/6/commits/7558a4cb5c0fd737be5c91cf40dbc83c1ff97b8a.  The tests no longer passed if I didn't explicitly call the Dispose on the new helper class.  If this makes sense I'll continue on with the remaining test classes but if not I can remove.  Thanks.

File: Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<Project>
+	<PropertyGroup Condition=" '$(TargetFramework)' == 'net48' or '$(TargetFramework)' == 'netcoreapp3.1'">
+		<EnableNETAnalyzers>true</EnableNETAnalyzers>
+	</PropertyGroup>
+	<PropertyGroup>
+		<TargetFrameworks>net6.0;netcoreapp3.1;net48;</TargetFrameworks>
```
Comment (eslutz) [inline Directory.Build.props]: @NeilSchiefer I just noticed, shouldn't you have `<LangVersion>latest</LangVersion>` and `<Nullable>enable</Nullable>` in your Directory.Build.props as part of the modernization?

Comment (NeilSchiefer) [reply Directory.Build.props]: Thanks for catching that.

**Review by JasonBock — APPROVED** [review]

File: RockLib.Reflection.Optimized/PropertyGetter.cs [hunk 2]
```diff
@@ -10,28 +10,32 @@ internal class PropertyGetter
...
+            Expression body = null!;
 
-            if (_property.GetMethod.IsStatic)
+            if (_property.GetMethod!.IsStatic)
+            {
                 body = Expression.Call(_property.GetMethod);
-            else
+            }
+            else if (_property.DeclaringType != null)
+            {
```
Comment (eslutz) [inline RockLib.Reflection.Optimized/PropertyGetter.cs:34]: I think this should be `is not null`.

File: RockLib.Reflection.Optimized/PropertyGetter.cs [hunk 3]
```diff
@@ -60,14 +64,23 @@ public void SetOptimizedFunc()
...
+                return;
+            }
 
             if (_property.GetMethod.IsStatic)
+            {
                 body = Expression.Call(_property.GetMethod);
-            else
+            }
+            else if (_property.DeclaringType != null)
```
Comment (eslutz) [inline RockLib.Reflection.Optimized/PropertyGetter.cs]: Same here.

File: RockLib.Reflection.Optimized/PropertyGetter.cs (hunk 2 above)
Comment (NeilSchiefer) [reply RockLib.Reflection.Optimized/PropertyGetter.cs:34]: 👍

File: RockLib.Reflection.Optimized/PropertyGetter.cs (hunk 3 above)
Comment (NeilSchiefer) [reply RockLib.Reflection.Optimized/PropertyGetter.cs]: 👍

**Review by eslutz — COMMENTED** [review]
I commented on a couple places where I saw `!= null` instead of `is not null`. It looks like it's in several places/files so you might want to just go through and for a find and replace for it.

**Comment by NeilSchiefer** [comment]
> I commented on a couple places where I saw `!= null` instead of `is not null`. It looks like it's in several places/files so you might want to just go through and for a find and replace for it.

I think I got them all thanks!

**Review by eslutz — APPROVED** [review]

**Pull request merged**